src/content/guides/multi-nginx/index.md

```diff
@@ -0,0 +1,81 @@
+---
+title: Running Multiple NGINX Ingress Controllers
+description: Deploy multiple NGINX ingress controllers in a Kubernetes cluster to separate different ingress traffic classes.
+date: 2020-08-05
+type: page
+tags: ["tutorial"]
+---
+
+# Multi-NGINX ingress controller support
+
+NGINX ingress controller handles [Ingress](https://kubernetes.io/docs/concepts/services-networking/ingress/) resources, routing the traffic from outside of the Kubernetes cluster to services within the cluster. Starting with App version 1.8.0 one can install multiple NGINX ingress controllers in same Kubernetes cluster. Here we cover common cases where having multiple NGINX ingress controllers can be useful.
+
+Most of the NGINX configuration options have NGINX wide defaults but can also be overriden on per Ingress resource level. In each of the covered multi-NGINX use cases, one essentially installs second NGINX with different NGINX global-only configuration, i.e. configuration that applies to all Ingress resources managed by that NGINX installation which cannot be customized on per-Ingress resource level.
+
+## Separating public from internal ingress traffic
+
+Having some components in multi-tierd architecture or back-office applications not accessible via public internet is a common requirement for security or performance reasons. This is how one can achieve it by using multi-NGINX ingress controller support:
+
+- Deploy one NGINX IC App using default settings, for making selected cluster services accessible via the public internet
+- Deploy second internal NGINX IC App using following user configuration overrides
+
+  - on AWS and Azure
+
+    ```yaml
+    controller:
+      ingressClass: "nginx-internal"
+      service:
+        public: false
+        subdomain: "ingress-internal"
+    ```
+
+  - on KVM
+
+    ```yaml
+    controller:
+      ingressClass: "nginx-internal"
+      service:
+        # enabled: true
+        public: false
+        subdomain: "ingress-internal"
+        # type: NodePort
+        nodePorts:
+          http: 31010
+          https: 31011
+    ```
+
+Each NGINX IC App installation has to have unique ingress class. Ingress resources can then declare which ingress controller should be handling their route definition by referencing appropriate ingress class in `kubernetes.io/ingress.class` annotation. Default NGINX IC App ingress class is `nginx`. In the above example we configure `nginx-internal` as the second NGINX IC App installation's ingress class.
```

Comment:
```suggestion
Each NGINX IC App installation has to have unique `ingressClass`. Ingress resources can then declare which ingress controller should be handling their route definition by referencing appropriate `ingressClass` in `kubernetes.io/ingress.class` annotation. Default NGINX IC App ingress class is `nginx`. In the above example we configure `nginx-internal` as the second NGINX IC App installation's ingress class.
```

Reply:
Ingress class is configured via `controller.ingressClass` config property in NGINX, but via `kubernetes.io/ingress.class` annotation in Ingress resources.

I was deliberately not referring to either of the two but to ingress class concept as such.

Things are evolving, in new k8s 1.18 https://kubernetes.io/docs/concepts/services-networking/ingress/#ingress-class
instead of annotation a IngressClass/ingressClassName got specified for use in Ingress resources.

Whatever the means are to specify ingress class (old or new), each NGINX will have to have a unique one associated with it, and Ingress resources will have to have be classified with one of them.